File: TopInterview150/27_remove_element/removeElement.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l, r = 0, len(nums) - 1

        k = 0

        while l <= r:
            if nums[r] == val:
                nums[r] = None
                r -= 1
                k += 1
            elif nums[l] == val:
                nums[l] = nums[r]
                nums[r] = None
                r -= 1
                l += 1
                k += 1

            else:
                l += 1

        return len(nums) - k

File: TopInterview150/27_remove_element/removeElement.ts
function removeElement(nums: number[], val: number): number {
  const count = new Array<number>(51).fill(0);

  for (let i of nums) {
    if (i !== val) {
      count[i]++;
    }
  }

  let pos = 0;

  for (let i = 0; i < count.length; i++) {
    while (count[i] > 0) {
      nums[pos] = i;
      count[i]--;
      pos++;
    }
  }

  return pos;
}
